src/com/dabomstew/pkrandom/constants/Gen4Constants.java

```diff
@@ -150,18 +154,166 @@
+                lifeOrb, toxicOrb, flameOrb, zoomLens, metronome, ironBall, laggingTail, destinyKnot,
                 blackSludge, icyRock, smoothRock, heatRock, dampRock, gripClaw, choiceScarf, stickyBarb, shedShell,
                 bigRoot, choiceSpecs, flamePlate, splashPlate, zapPlate, meadowPlate, iciclePlate, fistPlate,
                 toxicPlate, earthPlate, skyPlate, mindPlate, insectPlate, stonePlate, spookyPlate, dracoPlate,
-                dreadPlate, ironPlate, oddIncense, rockIncense, fullIncense, waveIncense, roseIncense, pureIncense,
+                dreadPlate, ironPlate, oddIncense, rockIncense, fullIncense, waveIncense, roseIncense,
                 razorClaw, razorFang));
         list.addAll(consumableHeldItems);
         return list;
     }
 
-    public static final Map<Integer,List<Integer>> abilityVariations = setupAbilityVariations();
+    // Moves (their move index) affected by items
+    // https://bulbapedia.bulbagarden.net/wiki/List_of_moves
+    public static final int absorb = 71, aquaRing = 392, bind = 20, bounce = 340, clamp = 128, dig = 91, dive = 291,
+            drainPunch = 409, dreamEater = 138, facade = 263, fireSpin = 83, fling = 374, fly = 19,
+            gigaDrain = 202, hail = 258, ingrain = 275, leechLife = 141, leechSeed = 73, lightScreen = 113,
+            magmaStorm = 463, megaDrain = 72, outrage = 200, psychoShift = 375, rainDance = 240, razorWind = 13,
+            reflect = 115, sandstorm = 201, sandTomb = 328, skullBash = 130, skyAttack = 143, solarBeam = 76,
+            sunnyDay = 241, switcheroo = 415, trick = 271, uproar = 253, whirlpool = 250, wrap = 35;
+    // Abilties with positive item synergy
+    // https://bulbapedia.bulbagarden.net/wiki/Ability#List_of_Abilities
+    public static final int guts = 62;
+
+    public static final List<Integer> generalPurposeConsumableItems = Collections.unmodifiableList(Arrays.asList(
+            cheriBerry, chestoBerry, pechaBerry, rawstBerry, aspearBerry, leppaBerry,
+            oranBerry, persimBerry, lumBerry, sitrusBerry, ganlonBerry, salacBerry,
+            // An NPC pokemon's nature is generated randomly with IVs during gameplay. Therefore, we include
+            // all these berries, even though they might confuse the opponent's pokemon if the nature is wrong.
+            figyBerry, wikiBerry, magoBerry, aguavBerry, iapapaBerry,
+            chilanBerry, apicotBerry, lansatBerry, starfBerry, enigmaBerry, micleBerry, custapBerry,
+            jabocaBerry, rowapBerry, berryJuice, whiteHerb, mentalHerb, focusSash));
+
+    public static final List<Integer> generalPurposeItems = Collections.unmodifiableList(Arrays.asList(
+            brightPowder, quickClaw, choiceBand, kingsRock, focusBand,
+            scopeLens, leftovers, shellBell, laxIncense, wideLens, expertBelt, lifeOrb, zoomLens,
+            laggingTail, destinyKnot, stickyBarb, shedShell, fullIncense, razorClaw, razorFang));
```

Comment:
Choice Band shouldn't be here for the same reason as Gen 3. Additionally, I'm not sure if I would call Lagging Tail "general purpose"

Reply:
Done. Removed laggingTail and fullIncense from general purpose and moved them to moveBoostingItems.